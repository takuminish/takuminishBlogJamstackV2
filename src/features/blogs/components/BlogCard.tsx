import Card from "@mui/material/Card";
import CardActionArea from "@mui/material/CardActionArea";
import CardContent from "@mui/material/CardContent";
import ValueWrapper from "./ValueWarapper";
import Typography from "@mui/material/Typography";
import React from "react";
import Box from "@mui/material/Box";
import Category from "../../../shared/components/Category";
import DateText from "./DateText";
import Tag from "../../../shared/components/Tag";
import { BlogMetaData } from "../types";
import { Link } from "gatsby";
import Button from "@mui/material/Button";

export type Props = {
  metaData: BlogMetaData;
};

function BlogCard(props: Props) {
  const { metaData } = props;

  return (
    <Card sx={{ backgroundColor: "#fffff" }}>
      <CardActionArea component={Link} to={metaData.id}>
        <CardContent>
          <ValueWrapper>
            <Typography variant="h6">{metaData.frontmatter.title}</Typography>
            <Typography variant="body2" color="text.secondary">
              {metaData.frontmatter.description}
            </Typography>
          </ValueWrapper>
          <ValueWrapper>
            <Category text={metaData.frontmatter.category} />
            <Box
              sx={{
                display: "flex",
                justifyContent: "start",
                gap: "5px",
              }}
            >
              {metaData.frontmatter.tags.map((tag) => (
                <Tag key={tag} text={tag} />
              ))}
            </Box>
          </ValueWrapper>
          <ValueWrapper>
            <DateText
              label="createdAt"
              date={new Date(metaData.frontmatter.date)}
            />
          </ValueWrapper>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}

export default BlogCard;
