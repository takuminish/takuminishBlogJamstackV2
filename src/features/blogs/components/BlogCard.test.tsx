import React from "react";
import { render } from "@testing-library/react";
import BlogCard, { Props } from "./BlogCard";

const createDOM = () => {
  const props: Props = {
    metaData: {
      id: "123",
      frontmatter: {
        date: "2020-10-31T13:37:00.000Z",
        description: "冪乗を高速に計算するアルゴリズム 計算量O(logn)",
        title: "【アルゴリズム】繰り返し二乗法",
        category: "アルゴリズム",
        tags: ["競プロ", "アルゴリズム"],
      },
    },
  };

  return render(<BlogCard metaData={props.metaData} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
