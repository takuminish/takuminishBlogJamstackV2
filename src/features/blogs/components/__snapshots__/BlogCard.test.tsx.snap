// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mumbnf-MuiPaper-root-MuiCard-root"
  >
    <a
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      href="/123"
      tabindex="0"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-l57rop"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            【アルゴリズム】繰り返し二乗法
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            冪乗を高速に計算するアルゴリズム 計算量O(logn)
          </p>
        </div>
        <div
          class="MuiBox-root css-l57rop"
        >
          <div
            class="MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1gtxin7-MuiSvgIcon-root"
              data-testid="FolderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
              />
            </svg>
            <span
              class="MuiBox-root css-9vtw2v"
            >
              アルゴリズム
            </span>
          </div>
          <div
            class="MuiBox-root css-1l50nvy"
          >
            <span
              class="MuiBox-root css-1kcf2o3"
            >
              競プロ
            </span>
            <span
              class="MuiBox-root css-1kcf2o3"
            >
              アルゴリズム
            </span>
          </div>
        </div>
        <div
          class="MuiBox-root css-l57rop"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            createdAt
            : 
            2020/10/31
          </p>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
